### PR TITLE
Fix the breaking changes in ONNX PR #58

### PR DESCRIPTION
**Pull request by houseroad**
(1) Regenerate the nanopb files,
(2) Add ModelProto
(3) Update the test expect files.

CC: @ezyang

### CODE REVIEW

**Comment by soumith** [comment]
AttributeError: module 'onnx' has no attribute 'ModelProto'

File: torch/csrc/onnx/onnx.h [hunk 1]
```diff
@@ -284,4 +277,28 @@ class GraphProto : public MicroProto<onnx_GraphProto> {
...
 };
 
+class ModelProto : public MicroProto<onnx_ModelProto> {
+private:
+  std::string producer_name;
+  std::string producer_version;
+  std::string domain;
+  std::string doc_string;
+  // graph is a static field in nanopb struct.
+  GraphProto graph;
```
Comment (ezyang) [inline torch/csrc/onnx/onnx.h]: There is a better way to handle this case, see `Note [Callback for nested messages]` in `torch/csrc/onnx/onnx.options`.

File: torch/csrc/onnx/onnx.h [hunk 2]
```diff
@@ -284,4 +277,28 @@ class GraphProto : public MicroProto<onnx_GraphProto> {
...
+    proto.producer_name = string(&producer_name, "pytorch");
+    // TODO: stop hard-coding this
+    proto.producer_version = string(&producer_version, "0.2");
+    proto.domain = string(&domain, "com.facebook");
+  }
+  void set_model_version(int64_t i) { proto.has_model_version = true; proto.model_version = i; }
+  void set_doc_string(const std::string& s) { proto.doc_string = string(&doc_string, s); }
+  GraphProto* mutable_graph() { return &graph; }
+  // set_graph should be called after encoding graph.
+  void set_graph() { proto.has_graph = true; proto.graph = graph.proto; }
```
Comment (ezyang) [inline torch/csrc/onnx/onnx.h]: And then this goes away, and we get an exact API compatibility with the Google Protobuf C++ interface :)

**Review by ezyang — CHANGES_REQUESTED** [review]
This is great, thank you! There is one improvement that can be made, but besides that, let's get this in!

**Comment by houseroad** [comment]
ModelProto.graph is changed to callback type. Thanks for the information, @ezyang .

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
Looks great, thank you!

**Pull request merged**